app/controllers/news_controller.rb

```diff
@@ -36,13 +40,46 @@ def show
 
 
   private
+  # An array of sections the user is filtering by, loaded by params
+  # with section ids
+  def set_filters
+    @filters = []
+    section_ids = Section.all.collect(&:id)
```

Comment:
I think `Section.all.collect(&:Id)` will load all sections from the database. You could probably do `Section.pluck(:id)` instead to run `SELECT id FROM sections`.